src/sentry/rules/actions/utils.py

```diff
@@ -126,7 +126,7 @@ def get_changed_data(
     if rule_data_before.get("environment_id") and not rule_data.get("environment_id"):
         environment = None
         try:
-            environment = Environment.objects.get(id=rule_data.get("environment_id"))
+            environment = Environment.objects.get(id=rule_data_before.get("environment_id"))
```

Comment:
these should be square brackets (same as above) -- we know it is present and non-None here already (part of why the type checker flagged it for me!)